game/menu.hh

```diff
@@ -95,18 +95,17 @@ public:
 	void moveToLast() { selection_stack.back() = menu_stack.back()->size() - 1; }
 
 	size_t curIndex() { return selection_stack.back(); }
-	MenuOption& current() { return menu_stack.back()->at(selection_stack.back()); }
-	MenuOption& back() { return root_options.back(); }
+	MenuOption& current() { return *(menu_stack.back())->at(selection_stack.back()); }
+	MenuOption& back() { return *(root_options->back()); }
 	const MenuOptions::const_iterator begin() const { return menu_stack.back()->begin(); }
 	const MenuOptions::const_iterator end() const { return menu_stack.back()->end(); }
-	const MenuOptions getOptions() const { return *menu_stack.back(); }
+	const MenuOptions& getOptions() const { return *menu_stack.back(); }
 
 	Dimensions dimensions;
 
 private:
-	MenuOptions root_options;
+	bool m_open;
+	std::shared_ptr<MenuOptions> root_options;
```

Comment:
std::shared_ptr<MenuOptions> root_options = nullptr; If you really want to explicitely show it was initialized